readme.md

```diff
@@ -113,6 +113,16 @@ function listItem(n: number) {
 </ul>
 ```
 
+### Custom Elements
```

Comment:
There already is a section called "Custom Elements" further down. I think using "Custom Elements" here is confusing, seeing as there is no custom element in the *resulting html*. Could we call it "Using a helper template like an element" instead (or something like that)?

Reply:
What do you think of this? It would appear right after the current "Helper Templates"

#### Using a helper template like an element

Want a helper component? Create a function that implements CustomElementHandler and you can call it like an HTML element. 

```typescript
interface CustomElementHandler {
    (attributes: Attributes | undefined, contents: string[]): string
}

function Button(attributes: Attributes | undefined, contents: string[]) {
    return <div><button type="button" class="original-class" {...attributes}>{contents}</button></div>;
}
    
<Button style="color:#f00">Button Text</Button>
```

Prints: 

```html
<div>
    <button type="button" class="original-class" style="color:#f00">Button Text</button>
</div>
```

Reply:
Wow cool! I didn't even think of using the `CustomElementHandler` interface. In that case we can also expose the interface with an `export`. That way you can add something like this to the readme: 

**Using a helper template like an element**

Want a to call your helper like an element? Create a function that implements `CustomElementHandler` and you can call it like an HTML element.

```typescript
import { CustomElementHandler } from 'typed-html';

function Button(attributes: Attributes | undefined, contents: string[]) {
    return <div><button type="button" class="original-class" {...attributes}>{contents}</button></div>;
}
// Or 
const Button: CustomElementHandler = (attributes, contents) => <div><button type="button" class="original-class" {...attributes}>{contents}</button></div>;
}

console.log(<Button style="color:#f00">Button Text</Button>);
```

Prints:

```html
<div>
    <button type="button" class="original-class" style="color:#f00">Button Text</button>
</div>
```